xaitk_saliency/impls/vis_sal_similarity/similarityscoring.py

```diff
@@ -0,0 +1,61 @@
+from xaitk_saliency import ImageSimilaritySaliencyMapGenerator
+from xaitk_saliency.utils.masking import weight_regions_by_scalar
+
+import numpy as np
+from sklearn.preprocessing import minmax_scale
+from sklearn.metrics.pairwise import euclidean_distances
```

Comment:
[scipy.spatial.distance.cdist ](https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.distance.cdist.html#scipy.spatial.distance.cdist) may be slightly more general and future-proof here. It allows for passing in of a metric argument, which can be any distance function, e.g. Euclidean distance, cosine distance, etc.

Reply:
Great idea, I added support for all distances supported by scipy. Thank you.

Reply:
Really minor point, but I would use the following import, and then just use `cdist()` everywhere else:

```python
from scipy.spatial.distance import cdist
```

Reply:
Done.